File: OneDrive/Desktop/COEXIST/FrontCoexist-final1Angular/FrontCoexist-final1/src/app/entity/Preference.ts
import {Day} from "./Day";
import {CarpoolingType} from "./CarpoolingType";
import {UserModel} from "./User.model";

export class Preference {
  preferenceID!: number;
  departureTime!: Date;
  longitudeDeparture!: string;
  latitudeDestination!: string;
  latitudeDeparture!: string;
  longitudeDestination!: string;
  availableSeats!: number;
  costPerSeat!: number;
  day!: string;
  time!: string;
  carpoolingType!: string;
  //user: UserModel;
}

File: OneDrive/Desktop/COEXIST/backcoexist-final1SpringBoot/src/main/java/tn/esprit/coexist/entity/Preference.java
package tn.esprit.coexist.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Data
public class Preference{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer preferenceID;
    private LocalDateTime DepartureTime;
    private String longitudeDeparture;
    private String latitudeDestination;

    private String latitudeDeparture;
    private String longitudeDestination;



    private  int AvailableSeats;
    private float CostPerSeat;
    @Enumerated(EnumType.STRING)
    private Day day;
    private  LocalTime time;
    @Enumerated(EnumType.STRING)
    private CarpoolingType carpoolingType;
    @OneToOne(mappedBy = "preference")

    private User user;

}
